### PR TITLE
DEV2-3197 breakdown editor context to dynamic context parts

### PR DESCRIPTION
**Pull request by yonip23**

### CODE REVIEW

**Comment by yonip23** [comment]
TODO: bump chat version once its released

**Review by yairco1990 — APPROVED** [review]
lgtm

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+) : EnrichingContextData {
+    // Used for serialization - do not remove
+    private val type: EnrichingContextType = EnrichingContextType.Diagnostics
+
+    companion object {
+        fun create(editor: Editor, project: Project): DiagnosticsContext? {
+            val visibleRange = getVisibleRange(editor) ?: return null
+            val highlights = mutableListOf<String>()
+
+            DaemonCodeAnalyzerImpl.processHighlights(
```
Comment (dimacodota) [inline Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt:23]: can you check if this can throw an exception ?

Comment (yonip23) [reply Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt:23]: it doesn't say so in the method declaration, but probably can throw an exception if you don't have read access or something.
anyway, if it throws an exception, we will send an error message to the chat and the chat knows how to handle it.
good enough IMO, wdyt?

**Review by dimacodota — APPROVED** [review]
lgtm

**Pull request merged**